plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt

```diff
@@ -64,14 +63,14 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
         }
 
         private fun renderModifiersList(modifiersListOwner: JKModifiersListOwner) {
-            val isOpenByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
+            val isOpenAndAbstractByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
                     (modifiersListOwner is JKMethod && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
+                    || (modifiersListOwner is JKField && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
             val hasOverrideModifier = modifiersListOwner
                 .safeAs<JKOtherModifiersOwner>()
                 ?.hasOtherModifier(OVERRIDE) == true
-
             fun Modifier.isRedundant(): Boolean = when {
-                this == OPEN && isOpenByDefault -> true
+                (this == OPEN || this == ABSTRACT) && isOpenAndAbstractByDefault -> true
```

Comment:
I'm a little confused. We have both "or" and "and" here. Which one is it? Maybe it's better to split this into two branches?

Reply:
the interface would be both open and abstract by default, so if it has either modifier we should remove it- i can separate it into two branches if that makes it clearer